lib/iris/_deprecation_helpers.py

```diff
@@ -24,10 +24,20 @@
 
 import warnings
 
+from iris.exceptions import IrisError
+
+
+class IrisDeprecation(IrisError, DeprecationWarning):
```

Comment:
I don't get why you're inheriting from `IrisError`.

1. It's not an error.
2. We don't need the inheritance and don't make use of it.